polling_stations/apps/data_importers/base_importers.py

```diff
@@ -124,7 +91,7 @@ def check_positive(value):
     def teardown(self, council):
         PollingStation.objects.filter(council=council).delete()
         PollingDistrict.objects.filter(council=council).delete()
-        ResidentialAddress.objects.filter(council=council).delete()
+        UprnToCouncil.objects.filter(lad=council.pk).update(polling_station_id="")
 
     def get_council(self, council_id):
         return Council.objects.get(identifiers__contains=[council_id])
```

Comment:
Where are you thinking of going next with this?

Reply:
I imagine changing the `council_id` [the gss code] to something like `council_identifier` and letting `council_id` always be a reg code. Essentially trying to make the code easier to read, by not referring to gss codes and reg codes as both 'council_id'

Reply:
Yes. That is a useful thing to do, or maybe even call them `gss_code` and `register_code` :open_mouth:

I'm going to chuck one other thing into the mix here to think about:

- The point of a register code is it identifies an organisation. If the boundary of St Albans Council changes subtly, say from https://mapit.mysociety.org/area/E07000100.html to https://mapit.mysociety.org/area/E07000240.html it is still the same organisation. It keeps the same identifier.
- The point of a GSS code is it identifies a polygon. If the boundary of St Albans Council changes subtly, say from https://mapit.mysociety.org/area/E07000100.html to https://mapit.mysociety.org/area/E07000240.html it is a new polygon. It gets a new identifier.

So a register code is conceptually correct and useful if the information we need to store is "The phone number for St Albans Council is 01727 819294". If the polygon changes, that fact is still true. However, if the information we want to store is "'115 WATLING STREET, AL2 2NN' is in St Albans" (essentially "this point is inside that polygon") it makes more sense to store that information in terms of a GSS code because changing the polygon changes the fact. In moving to a situation where the `uprntocouncil` table stores register codes (not in this PR) we've moved to a situation where we're using register codes to store "this point is inside that polygon" which becomes _super-confusing_ when minor boundary changes happen because it becomes somewhat ambiguous whether the fact in the DB is "115 WATLING STREET is in E07000100" or "115 WATLING STREET is in E07000240". If you're going to try and clean up when to use a register code and when to use a GSS code as a future piece of work, I'd encourage you to think about this principle in deciding when and why we use a particular code for a particular purpose:

- If changing the polygon changes the fact/data point, store the fact against a GSS code
- If changing the polygon doesn't change the fact/data point, store the fact against a register code

At the moment this principle is violated, but I think following that rule probably leads you to using the right code in the right situation which should then be more intuitive and make problems more obvious. This is something we currently do well in Every Election, but poorly in WDIV.

Reply:
> Yes. That is a useful thing to do, or maybe even call them gss_code and register_code open_mouth

Indeed!

>    - If changing the polygon changes the fact/data point, store the fact against a GSS code
>    - If changing the polygon doesn't change the fact/data point, store the fact against a register code

These principles make a lot of sense. Feels like reg codes should be the pk in the councils table, but now the UprnToCouncil table needs to be fixed, and we need a way of knowing which boundary (/gss code) is currently in the `council.area` field. I'll make an issue to look at it.